app/submitter/converter.py

```diff
@@ -82,14 +83,30 @@ def convert_answers_to_data(answer_store, questionnaire_schema):
     for answer in answer_store.answers:
         item = questionnaire_schema.get_item_by_id(answer['answer_id'])
         value = answer['value']
+
         if item is not None and value is not None:
-            if item.code not in data:
-                data[item.code] = _encode_value(value)
+            if not isinstance(item, CheckboxAnswer) or any('q_code' not in option for option in item.options):
```

Comment:
I didn't that initially, however I changed it back for a couple of reason. 1. most surveys won't use q_codes in their checkboxes and therefore I felt this was more efficient 2. I had concerns over a missed q_code in a checkbox when the others had them. I originally had a safety net in place to catch this, however the other box turned to non mandatory made a mess of this, I therefore decided to make sure every option had a q_code and reverse the condition